From 54a75a63047952e3e09320587212ca9e7b2b8c1f Mon Sep 17 00:00:00 2001
From: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Date: Fri, 25 Jan 2019 12:34:17 +0530
Subject: [PATCH 0140/1852] ASoC: xlnx: enable multi sample rate support for
 HDMI sound card

commit 1c2c85a7bbad9230fb142dfa4a68ab1db8866763 from
https://github.com/Xilinx/linux-xlnx.git

PAtch enables HDMI sound card with multiple sampling rates for playback.
Added a proper mclk ratio based on ACR IP version used.
The ratio will change with a corresponding change in ACR IP.

Signed-off-by: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_pl_snd_card.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/sound/soc/xilinx/xlnx_pl_snd_card.c b/sound/soc/xilinx/xlnx_pl_snd_card.c
index 5ba37dcbde1e..9950ff9b38b5 100644
--- a/sound/soc/xilinx/xlnx_pl_snd_card.c
+++ b/sound/soc/xilinx/xlnx_pl_snd_card.c
@@ -87,13 +87,11 @@ static int xlnx_hdmi_card_hw_params(struct snd_pcm_substream *substream,
 	case 32000:
 	case 44100:
 	case 48000:
+	case 88200:
 	case 96000:
 	case 176400:
 	case 192000:
-		prv->mclk_ratio = 768;
-		break;
-	case 88200:
-		prv->mclk_ratio = 192;
+		prv->mclk_ratio = 512;
 		break;
 	default:
 		return -EINVAL;
@@ -227,6 +225,7 @@ static struct snd_soc_dai_link xlnx_snd_dai[][XLNX_MAX_PATHS] = {
 		{
 			.name = "xilinx-hdmi-playback",
 			SND_SOC_DAILINK_REG(xlnx_hdmi_tx),
+			.ops = &xlnx_hdmi_card_ops,
 		},
 		{
 			.name = "xilinx-hdmi-capture",
-- 
2.31.1

